### PR TITLE
Add Remove function

### PR DESCRIPTION
**Pull request by chocolacula**
I believe checking index and bounds is better than keep it without the check and panicking with a bit weird error.

### CODE REVIEW

**Comment by chocolacula** [comment]
@elliotchance any thoughts?

**Comment by chocolacula** [comment]
> I don't like the way it uses an output parameter. It should return a new slice not containing the element. Also, Delete a more common term used by languages when deleting something by it's key. And finally, are you sure this works with a single element? It looks like it would panic. 

Done. One element will be deleted correctly. Added a test for that case.

File: v2/delete.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package pie
+
+// Removes element at index idx from input slice, returns resulting slice.
+// If the index out of bounds, returns unchanged input slice.
+func Delete[T any](ss []T, idx int) []T {
```
Comment (elliotchance) [inline v2/delete.go:5]: This also needs to be registered against the chainable types: https://github.com/elliotchance/pie/blob/master/v2/of.go, https://github.com/elliotchance/pie/blob/master/v2/of_numeric.go and https://github.com/elliotchance/pie/blob/master/v2/of_ordered.go

Comment (elliotchance) [reply v2/delete.go:5]: It also makes sense to me to allow zero or more indexes:

```
.Delete(3, 4, 5)
.Delete(bunchOfIndexes...)
```

Although this is a little more complicated, so I'm happy for that to come later if needed.

**Review by elliotchance — CHANGES_REQUESTED** [review]

**Comment by chocolacula** [comment]
I tried a few approaches and chose the best one. I added a little benchmark:
```
BenchmarkDelete-8                   3372           355714 ns/op          8003676 B/op      3 allocs/op
BenchmarkDelete-8                   2890           357530 ns/op          8003675 B/op      3 allocs/op
BenchmarkDeleteShort-8          14604861           80.06 ns/op           176 B/op          3 allocs/op
BenchmarkDeleteShort-8          14294144           80.04 ns/op           176 B/op          3 allocs/op
BenchmarkDeleteSet-8                 184           6437496 ns/op         8003588 B/op      1 allocs/op
BenchmarkDeleteSet-8                 184           6431757 ns/op         8003589 B/op      1 allocs/op
BenchmarkDeleteSetShort-8        8599725           137.4 ns/op           128 B/op          1 allocs/op
BenchmarkDeleteSetShort-8        8649344           137.3 ns/op           128 B/op          1 allocs/op
```

You can find them in separate branch in [delete_test.go](https://github.com/chocolacula/pie/blob/delete_benchmarks/v2/delete_test.go)

**Comment by chocolacula** [comment]
@elliotchance please review

**Review by elliotchance — APPROVED** [review]

**Pull request merged**